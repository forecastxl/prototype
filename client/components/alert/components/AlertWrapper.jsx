import muiThemeable from 'material-ui/styles/muiThemeable'
import styled, { keyframes } from 'styled-components'
import styles from '../../../styles'

const fadeIn = keyframes`
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
`

export const AlertWrapper = styled.div`
  background: ${styles.colors.error};
  font-weight: bold;
  font-size: 0.75rem;
  color: ${props => props.muiTheme.palette.alternateTextColor};
  padding: 0.25rem 0.5rem;
  border-radius: 3px;
  animation: ${fadeIn} 200ms linear;
`

export default muiThemeable()(AlertWrapper)
