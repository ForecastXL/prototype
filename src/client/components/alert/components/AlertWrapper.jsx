import { oneOf } from 'prop-types'
import styled, { keyframes } from 'styled-components'
import { boxShadow, colors } from '../../../styles'

const fadeIn = keyframes`
  from { opacity: 0; }
  to { opacity: 1; }
`

const setBackground = props => {
  const map = {
    error: colors.red,
    warning: colors.yellow,
    info: colors.blue,
    success: colors.green
  }

  return map[props.type]
}

const setColor = props => {
  const map = {
    error: colors.darkRed,
    warning: colors.darkYellow,
    info: colors.darkBlue,
    success: colors.darkGreen
  }

  return map[props.type]
}

const AlertWrapper = styled.div`
  align-items: center;
  animation: ${fadeIn} 200ms linear;
  background: ${setBackground};
  border-radius: 3px;
  box-shadow: ${boxShadow};
  color: ${setColor};
  display: flex;
  margin-bottom: 0.5rem;
  padding: 1rem 0.5rem;
`

AlertWrapper.propTypes = {
  type: oneOf(['error', 'warning', 'info', 'success']).isRequired
}

export default AlertWrapper
