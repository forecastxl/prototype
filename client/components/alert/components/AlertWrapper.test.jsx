import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { AlertWrapper } from './AlertWrapper'

jest.mock('material-ui')

describe('<AlertWrapper />', () => {
  it('renders correctly', () => {
    const theme = {
      palette: {
        alternateTextColor: 'red'
      }
    }
    const wrapper = shallow(<AlertWrapper muiTheme={theme} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
