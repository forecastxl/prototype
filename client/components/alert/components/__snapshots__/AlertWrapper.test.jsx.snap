exports[`<AlertWrapper /> renders correctly 1`] = `
<div
  className="iUyTNb" />
`;
